readthedocs/notifications/notification.py

```diff
@@ -84,3 +87,74 @@ def send(self):
         avoided.
         """
         send_notification(self.request, self)
+
+
+class SiteNotification(Notification):
+
+    """
+    Simple notification to show *only* on site messages.
+
+    ``success_message`` and ``failure_message`` can be a simple string or a
+    dictionary with different messages depending on the reason of the failure /
+    success. The message is selected by using ``message_key`` to get the proper
+    value.
+
+    The notification is tied to the ``user`` and it could be sticky, persistent
+    or normal --this depends on the ``success_level`` and ``failure_level``.
+
+    .. note::
+
+        ``send_email`` is forced to False to not send accidental emails when
+        only a simple site notification is needed.
+    """
+
+    send_email = False
+
+    success_message = None
+    failure_message = None
+
+    success_level = constants.SUCCESS_NON_PERSISTENT
+    failure_level = constants.ERROR_NON_PERSISTENT
+
+    def __init__(
+            self, user, success, message_key=None, context_object=None,
```

Comment:
Ha! I had `reason` and I changed to `message_key` because I wasn't sure that `reason` was clear enough :)

But yeah, `reason` sounds better. I will change it back.